ansi supports color tags split across lines. Also, better rainbows with whitespace

thomasmitchell opened the pull request:
Fixes #4 
Added the flag to make the `.` regexp operator match the newline character.

Also bundled in: rainbows don't waste colors on whitespace.

Commit:
ansi library can now support color tags split across lines

jhunt commented:
Can you add tests that both confirm the original bug and verify the proposed fix?

jhunt reviewed (changes requested):
Can you add tests that both confirm the original bug and verify the proposed fix?

thomasmitchell committed:
tests for multiline parsing. rainbow powers++

Rainbowization no longer wastes colors in its sequence on whitespace characters that can't be bothered to display them.

thomasmitchell committed:
another rainbow test

thomasmmitchell commented:
Don't like to put two separate things on one ticket, but just trying to avoid the PR overhead. Any issues?

jhunt commented:
In the future, try to keep to one fix/feature per PR.
